cfme/tests/intelligence/reports/test_reports.py

```diff
@@ -239,3 +243,53 @@ def test_report_edit_secondary_display_filter(appliance, filter_report, soft_ass
         view.report_info.secondary_filter.read() == secondary_filter,
         "Secondary Filter did not match.",
     )
+
+
+@test_requirements.report
+@pytest.mark.tier(1)
+@pytest.mark.meta(server_roles="+notifier", automates=[1677839])
+@pytest.mark.provider([InfraProvider], selector=ONE_PER_CATEGORY)
+def test_send_text_custom_report_with_long_condition(
+    appliance, setup_provider, smtp_test, soft_assert, request, get_report
+):
+    """
+    Polarion:
+        assignee: pvala
+        casecomponent: Reporting
+        caseimportance: medium
+        initialEstimate: 1/3h
+        setup:
+            1. Create a report containing 1 or 2 columns
+                and add a report filter with a long condition.(Refer BZ for more detail)
+            2. Create a schedule for the report and check send_txt.
+        testSteps:
+            1. Queue the schedule and monitor evm log.
+        expectedResults:
+            1. There should be no error in the log and report must be sent successfully.
+
+    Bugzilla:
+        1677839
+    """
+    report = get_report("long_condition_report", "test_long_condition_report")
+    data = {
+        "timer": {"hour": "12", "minute": "10"},
+        "email": {"to_emails": "test@example.com"},
+        "email_options": {"send_if_empty": True, "send_txt": True},
+    }
+    schedule = report.create_schedule(**data)
+
+    # prepare LogValidator
+    log = LogValidator(
+        "/var/www/miq/vmdb/log/evm.log", failure_patterns=[".*negative argument.*"]
+    )
+
+    log.start_monitoring()
+    schedule.queue()
+
+    # assert that the mail was sent
+    assert (
+        len(smtp_test.wait_for_emails(wait=200, to_address=data["email"]["to_emails"]))
+        == 1
+    )
+    # assert that the pattern was not found in the logs
+    soft_assert(log.validate(), "Found error message in the logs.")
```

Comment:
Q: why `soft_assert`? I don't think you want to continue even if test fails here.